utils/src/main/scala/za/co/absa/enceladus/utils/broadcast/BroadcastUtils.scala

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.broadcast
+
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.ScalaReflection
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.functions.expr
+import za.co.absa.enceladus.utils.error.{ErrorMessage, Mapping}
+
+object BroadcastUtils {
+
+  def broadcastMappingTable(localMappingTable: LocalMappingTable)(implicit spark: SparkSession): Broadcast[LocalMappingTable] = {
+    spark.sparkContext.broadcast(localMappingTable)
+  }
+
+  /**
+    * Converts a Spark expression to an actual value that can be used inside a UDF.
+    *
+    * This is used to convert a default value provided by a user as a part of mapping rule definition,
+    * to a value that can be used inside a mapping UDF.
+    *
+    * The returned value is one of primitives, or a Row for structs, or an array of Row(s) for an array of struct.
+    *
+    * @param expression A Spark expression.
+    * @return A UDF that maps joins keys to a target attribute value.
+    */
+  def getValueOfSparkExpression(expression: String)(implicit spark: SparkSession): Any = {
+    import spark.implicits._
+
+    List(1).toDF()
+      .withColumn("a", expr(expression))
+      .collect()(0)(1)
```

Comment:
This might need a little more explanation about how it works. Not in the scaladoc